generators/php/codegen/src/ast/MergeArrays.ts

```diff
@@ -0,0 +1,53 @@
+import { AbstractAstNode } from "@fern-api/base-generator";
+
+import { codeblock } from "../php";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { writeArguments } from "./utils/writeArguments";
+
+export declare namespace MergeArrays {
```

Comment:
I can see the value of this use case, but the types are a little difficult to follow. Is there any way we can reduce this down to only the use case that we're currently using, i.e.

```typescript
php.mergeArrays(`$this->${this.context.getClientOptionsName()}`, {
  ref: php.variable(this.context.getRequestOptionsName()),
  fallback: "[]"
})
```

If not, then this is likely fine as-is - just thinking of ways to reduce complexity and potentially remove the need for the new `writer.writeNodeOrString` function entirely.

Reply:
If we don't accept strings, we'll have to wrap it in php.codeBlock() which is fine but I like reducing the friction.
I'm assuming we'll use this `mergeArrays` in a bunch of places.

I'm also down to just remove it and construct the merge call manually everywhere.